### PR TITLE
Update CSS token per new design system theming capability

### PR DESCRIPTION
**Pull request by irma-kurnia-alaska**
# Alaska Airlines Pull Request

**Fixes:** #50 

## Summary:

- Update the design tokens to reflect the new theming conventions regarding tokens from the AuroDesignTokens and JetstreamDesignTokens repo.

## Type of change:

Please delete options that are not relevant.

- [x] Revision of an existing capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -10,8 +10,9 @@
       href="../node_modules/prismjs/themes/prism.css"
     />
     <link rel="stylesheet" href="https://unpkg.com/@alaskaairux/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
-    <link rel="stylesheet" type="text/css" href="https://unpkg.com/@alaskaairux/webcorestylesheets@latest/dist/demoWrapper.css" />
-    <link rel="stylesheet" type="text/css" href="https://unpkg.com/@alaskaairux/webcorestylesheets@latest/dist/elementDemoStyles.css" />
+    <link rel="stylesheet" href="https://unpkg.com/@aurodesignsystem/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
+    <link rel="stylesheet" type="text/css" href="https://unpkg.com/@aurodesignsystem/webcorestylesheets@latest/dist/demoWrapper.css" />
+    <link rel="stylesheet" type="text/css" href="https://unpkg.com/@aurodesignsystem/webcorestylesheets@latest/dist/elementDemoStyles.css" />
```
Comment (blackfalcon) [inline demo/index.html:15]: Please update to mirror the following 

https://github.com/AlaskaAirlines/auro-avatar/blob/main/demo/index.html#L12-L17

File: package.json [hunk 2]
```diff
@@ -23,14 +23,16 @@
     "lit-element": "^2.5.1"
   },
   "peerDependencies": {
-    "@alaskaairux/design-tokens": "^3.0.0",
-    "@alaskaairux/webcorestylesheets": "^3.0.0",
+    "@alaskaairux/design-tokens": "^3.3.0",
```
Comment (blackfalcon) [inline package.json:26]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -23,14 +23,16 @@
...
+    "@aurodesignsystem/design-tokens": "^4.1.1",
+    "@aurodesignsystem/webcorestylesheets": "^5.0.2",
     "@webcomponents/webcomponentsjs": "^2.4.4",
     "focus-visible": "^5.1.0"
   },
   "devDependencies": {
     "@alaskaairux/design-tokens": "^3.3.0",
```
Comment (blackfalcon) [inline package.json:33]:
```suggestion
```

File: src/style.scss [hunk 4]
```diff
@@ -4,21 +4,24 @@
 // ---------------------------------------------------------------------
 
 // Import Auro tokens
-@import './node_modules/@alaskaairux/design-tokens/dist/tokens/SCSSVariables';
-@import './node_modules/@alaskaairux/design-tokens/dist/tokens/SassCustomProperties';
-@import './node_modules/@alaskaairux/design-tokens/dist/tokens/SCSSVariableMap';
+@import './../node_modules/@aurodesignsystem/design-tokens/dist/tokens/SCSSVariables';
+@import './../node_modules/@aurodesignsystem/design-tokens/dist/tokens/SassCustomProperties';
```
Comment (blackfalcon) [inline src/style.scss:8]:
```suggestion
```

File: src/style.scss [hunk 5]
```diff
@@ -4,21 +4,24 @@
...
+@import './../node_modules/@aurodesignsystem/design-tokens/dist/tokens/SCSSVariableMap';
 
-@import './node_modules/@alaskaairux/webcorestylesheets/dist/breakpoints';
-@import './node_modules/@alaskaairux/webcorestylesheets/dist/core';
+// Support for fallback values
+@import './../node_modules/@aurodesignsystem/design-tokens/dist/tokens/SCSSVariables.scss';
```
Comment (blackfalcon) [inline src/style.scss:12]:
```suggestion
```

File: package.json [hunk 6]
```diff
@@ -81,7 +83,8 @@
     "wc-sass-render": "^1.2.3",
```
Comment (blackfalcon) [inline package.json:83]: Update to 1.4

File: package.json [hunk 7]
```diff
@@ -81,7 +83,8 @@
...
     "web-component-analyzer": "^1.1.6",
     "webpack-merge": "^5.8.0",
-    "whatwg-fetch": "^3.6.2"
+    "whatwg-fetch": "^3.6.2",
+    "sass": "^1.49.9"
```
Comment (blackfalcon) [inline package.json:87]: Update to the following 

https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/package.json#L74-L78

File: package.json [hunk 8]
```diff
@@ -110,11 +113,12 @@
     "test": "karma start --coverage",
     "test:watch": "karma start --auto-watch=true --single-run=false",
     "test:ci": "npm-run-all test lint",
-    "sassBuild": "npm-run-all sassBuild:demo sassBuild:component postCss:component sassRender",
-    "sassBuild:dev": "npm-run-all sassBuild:demo sassBuild:component postCss:component sassRender",
-    "sassBuild:demo": "node-sass ./demo/sass/ --output ./demo/css/",
```
Comment (blackfalcon) [inline package.json:115]: why is this removed?

File: package.json [hunk 9]
```diff
@@ -110,11 +113,12 @@
...
-    "sassBuild:component": "node-sass ./src/ --output ./src/",
+    "sassBuild": "npm-run-all sassBuild:component postCss:component sassRender",
+    "sassBuild:dev": "npm-run-all sassBuild:component postCss:component sassRender",
+    "sassBuild:component": "sass --no-source-map src:src",
     "sassBuild:watch": "nodemon -e scss --watch src --exec npm run sassBuild:dev",
+    "sassBuild:debug": "sass —source-map scss/index.scss docs/index.css",
```
Comment (blackfalcon) [inline package.json:120]: Why is this here?

File: src/style.scss (hunk 4 above)
Comment (blackfalcon) [inline src/style.scss:8]:
```suggestion
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (fajar-apri-alaska) [reply package.json:83]: `wc-sass-render@1.4.0` doesn't include node-sass in their [package dependency](https://github.com/tristanMatthias/wc-sass-render/blob/1.4.0/package.json), meanwhile this component `npm run sassBuild` script use `node-sass` CLI, so I am not sure we should go with this, or there is something else that I'm missing.
Please advise, @blackfalcon.

File: demo/index.html (hunk 1 above)
Comment (irma-kurnia-alaska) [reply demo/index.html:15]: I will updated. thank

File: package.json (hunk 2 above)
Comment (irma-kurnia-alaska) [reply package.json:26]: I will updated. thank

File: package.json (hunk 3 above)
Comment (irma-kurnia-alaska) [reply package.json:33]: I can delete this because this have dependencies with webcorestyles using at demo/styles.scss in line 125
&.focus-visible { @include auro_focus-hyperlink(sass); }
using https://github.com/import './node_modules/@alaskaairux/webcorestylesheets/dist/essentials'; and https://github.com/import './node_modules/@alaskaairux/design-tokens/dist/tokens/SCSSVariables'; when I change into the new version, it's have error when build.

File: src/style.scss (hunk 4 above)
Comment (irma-kurnia-alaska) [reply src/style.scss:8]: I will updated

File: src/style.scss (hunk 5 above)
Comment (irma-kurnia-alaska) [reply src/style.scss:12]: i will updated

File: package.json (hunk 8 above)
Comment (irma-kurnia-alaska) [reply package.json:115]: I will revert back.

File: package.json (hunk 9 above)
Comment (irma-kurnia-alaska) [reply package.json:120]: I will revert back.

File: src/style.scss (hunk 4 above)
Comment (irma-kurnia-alaska) [reply src/style.scss:8]: i will updated

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
See comments in the PR

File: package.json (hunk 2 above)
Comment (fajar-apri-alaska) [reply package.json]: For your information, I'm also keeping this css with the old `@alaskaairux/design-tokens` in the other component that have this because I'm afraid that would broke the UI.

**Comment by blackfalcon** [comment]
This PR is getting increasingly complicated since we are asking for newer dependency updates and this repo is still very legacy. 

I strongly recommend that we update the whole repo and the updated token/WCSS work would come along with that. 

I am attempting to update a script that automates the repo upgrade process to remove the manual steps used recently.

**Comment by fajar-apri-alaska** [comment]
> This PR is getting increasingly complicated since we are asking for newer dependency updates and this repo is still very legacy.
> 
> I strongly recommend that we update the whole repo and the updated token/WCSS work would come along with that.
> 
> I am attempting to update a script that automates the repo upgrade process to remove the manual steps used recently.

Sure, Let us know when the ticket is ready for us to be worked on.

**Comment by jordanjones243** [comment]
This PR is being closed, as this work is being handled in another PR:
- https://github.com/AlaskaAirlines/auro-popover/pull/56
